File: Sep10/Lecture6Examples/Example22/MainActivity.java
package com.varunshankar.example22;


import androidx.fragment.app.FragmentTransaction;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Create and populate our custom list
        CustomListData customListData = new CustomListData(1000);

        //Put this into a bundle
        Bundle fragmentBundle = new Bundle();
        fragmentBundle.putParcelable("item_list",customListData);

        //Create the fragment
        MasterListFragment masterListFragment = new MasterListFragment();

        //Pass data to the fragment
        masterListFragment.setArguments(fragmentBundle);

        //Replace the fragment container
        FragmentTransaction fTrans = getSupportFragmentManager().beginTransaction();
        fTrans.replace(R.id.fl_frag_masterlist, masterListFragment, "frag_masterList");
        fTrans.commit();

    }
}

File: Sep10/Example22/app/src/main/java/com/varunshankar/example22/MainActivity.kt
package com.varunshankar.example22

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //Create and populate our custom list
        val customListData = CustomListData(1000)

        //Put this into a bundle
        val fragmentBundle = Bundle()
        fragmentBundle.putParcelable("item_list", customListData)

        //Create the fragment
        val masterListFragment = MasterListFragment()

        //Pass data to the fragment
        masterListFragment.arguments = fragmentBundle

        //Replace the fragment container
        val fTrans = supportFragmentManager.beginTransaction()
        fTrans.replace(R.id.fl_frag_masterlist, masterListFragment, "frag_masterList")
        fTrans.commit()
    }
}